### ISSUE TITLE
Improve and fix accessibility issues in Modal block

### ISSUE THREAD

**Issue by mauteri** [GatherPress/gatherpress#1041]
### Describe the bug

Good example of proper [accessibility for modal](https://designsystem.digital.gov/components/modal/).

I met with Gen Herres to review accessibility of new blocks. Two improvements can be made to the Modal block that we discovered.

* After clicking escape, close, or clicking outside the modal which closes it, focus should return to the button that original opened the modal. Right now, focus is lost.
* After clicking "Attending," the modal does not close (which is correct), however, focus should be on "Close" as that's the next logical step (if close button exists).

### Steps to Reproduce

To come...

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct
